File: MyActivity_1/app/src/main/java/com/example/myactivity_1/MainActivity.java
package com.example.myactivity_1;


// 기본적으로 가져다 쓸 것들을 가지고 온다.
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Log.d("Tag","onCreate 호출");
    }

    @Override
    protected void onStart() {
        super.onStart();

        Log.d("Tag","onStart 호출");

    }

    @Override
    protected void onResume() {
        super.onResume();

        Log.d("Tag","onResume 호출");
    }

    @Override
    protected void onPause() {
        super.onPause();

        Log.d("Tag","onPause 호출");
    }

    @Override
    protected void onStop() {
        super.onStop();

        Log.d("Tag","onStop 호출");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();

        Log.d("Tag","onDestroy 호출");
    }

}

File: Kotlin_Lab/week10_4/app/src/main/java/com/example/week10_4/MainActivity.kt
package com.example.week10_4

import android.content.Intent
import android.content.SharedPreferences
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import androidx.core.content.edit
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    lateinit var option: SharedPreferences
    lateinit var userInfo: SharedPreferences

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        option = getSharedPreferences("option", MODE_PRIVATE)
        userInfo = getSharedPreferences("user_info", MODE_PRIVATE)

        vBtnSave.setOnClickListener {
            userInfo.edit {
                putString("name", "${vEditName.text}")
                vEditName.setText("")
            }
            Toast.makeText(this, "저장되었습니다", Toast.LENGTH_SHORT).show()
        }

        vSwitchAlarm.setOnCheckedChangeListener { compoundButton, b ->
            option.edit {
                putBoolean("alarm", vSwitchAlarm.isChecked)
            }
        }

        vBtnMoveActivity.setOnClickListener {
            val intent = Intent(this, MainActivity2::class.java)
            startActivity(intent)
        }

    }

    override fun onStart() {
        savedValueLoad()
        super.onStart()
    }

    private fun savedValueLoad() {
        vEditName.setText(userInfo.getString("name", ""))

        vSwitchAlarm.isChecked = option.getBoolean("alarm", false)
    }
}